src/components/canvas-tools/geometry-tool.tsx

```diff
@@ -292,15 +292,56 @@ class GeometryToolComponentImpl extends BaseComponent<IProps, IState> {
   private handleCreatePolygon = (polygon: JXG.Polygon) => {
 
     const handlePointerDown = (evt: any) => {
-      // console.log(`Polygon handlePointerDown!`);
+      each(polygon.ancestors, point => {
+        const pt = point as JXG.Point;
+        this.dragPts[pt.id] = { initial: cloneDeep(pt.coords) };
+      });
     };
 
     const handleDrag = (evt: any) => {
-      // console.log(`Polygon handleDrag!`);
+      each(polygon.ancestors, point => {
+        const pt = point as JXG.Point;
+        if (!this.dragPts[pt.id]) {
```

Comment:
Theoretically that's true, but since we have no control over whether the browser will send the right events in the expected order in all circumstances, or whether JSXGraph will handle/dispatch the events properly in all circumstances, a dose of caution seems warranted.

Reply:
Makes sense!